portfolio/src/main/webapp/script.js

```diff
@@ -1,28 +1,28 @@
-// Copyright 2019 Google LLC
-//
-// Licensed under the Apache License, Version 2.0 (the "License");
-// you may not use this file except in compliance with the License.
-// You may obtain a copy of the License at
-//
-//     https://www.apache.org/licenses/LICENSE-2.0
-//
-// Unless required by applicable law or agreed to in writing, software
-// distributed under the License is distributed on an "AS IS" BASIS,
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and
-// limitations under the License.
+// Author: Jonathan Chavez - tabaresj@google.com
+var isTerminalOpen = false;
 
-/**
- * Adds a random greeting to the page.
- */
-function addRandomGreeting() {
-  const greetings =
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
+window.addEventListener("keydown", keyboardHandler);
 
-  // Pick a random greeting.
-  const greeting = greetings[Math.floor(Math.random() * greetings.length)];
 
-  // Add it to the page.
-  const greetingContainer = document.getElementById('greeting-container');
-  greetingContainer.innerText = greeting;
+function keyboardHandler(e){
+  e = e || window.event;
+  console.log(String.fromCharCode(e.keyCode))
+  if(String.fromCharCode(e.keyCode) === 'T'){
+    if(!isTerminalOpen){
+      openTerminal();
+      isTerminalOpen = true;
+    }
+    else{
+      closeTerminal();
+      isTerminalOpen = false;
+    }
+  }
 }
+
+function openTerminal(){
+  document.getElementById("terminal").style.visibility = "visible";
```

Comment:
From my experience, visibility is usually not suitable for this type of cases, since it just hides the element, but it's still there.

I recommend you to play with:

`display: initial;` and `display: none;`

This totally removes DOM events from the element selected. The other way, there could be interactive issues with future elements.

Reply:
Cool, thanks!